### PR TITLE
Refactored drop-in related route line component

### PR DESCRIPTION
**Pull request by cafesilencio**
### Description
Refactored drop-in related route line component to facilitate customizing route line options as well as implementing the vanishing route line feature.

### CODE REVIEW

File: libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/DropInNavigationView.kt [hunk 1]
```diff
@@ -67,6 +68,10 @@ class DropInNavigationView @JvmOverloads constructor(
         navigationContext.uiBinders = navigationUIBinders
     }
 
+    fun customize(options: MapboxRouteLineOptions) {
```
Comment (kmadsen) [inline libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/DropInNavigationView.kt:71]: nit: Will you put a comment here? Pretending it's a public api so we have less docs to write

File: libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/DropInNavigationViewContext.kt [hunk 2]
```diff
@@ -13,7 +15,11 @@ internal class DropInNavigationViewContext(
     val lifecycleOwner: LifecycleOwner,
     val viewModel: DropInNavigationViewModel,
     val mapView: MapView,
-    val viewGroup: ViewGroup,
```
Comment (kmadsen) [inline libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/DropInNavigationViewContext.kt:16]: it's a feature https://kotlinlang.org/docs/coding-conventions.html#trailing-commas

"It makes version-control diffs cleaner" :)

File: libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/routeline/DropInRouteLine.kt [hunk 3]
```diff
@@ -149,4 +164,8 @@ internal class DropInRouteLine(
             }
         )
     }
+
+    companion object {
```
Comment (kmadsen) [inline libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/routeline/DropInRouteLine.kt:168]:
```suggestion
    private companion object {
```

**Review by kmadsen — APPROVED** [review]
Only nits, looks good!

**Review by abhishek1508 — COMMENTED** [review]
This looks good as it abstracts out fetching route from `DropInRouteLine` component. Only thing I am trying to wrap my head around is why do we need this component in `FreeDriveScreenBinder`? It should only be in `RoutePreviewScreenBinder` and `ActiveGuidanceScreenBinder`.

/cc @kmadsen @cafesilencio

**Comment by cafesilencio** [comment]
> This looks good as it abstracts out fetching route from `DropInRouteLine` component. Only thing I am trying to wrap my head around is why do we need this component in `FreeDriveScreenBinder`? It should only be in `RoutePreviewScreenBinder` and `ActiveGuidanceScreenBinder`.
> 

It was always there. The work here is to expose the route line options externally so they can be controlled by external developers.  The work to refactor the route fetching will be in a different PR.  If you're in free drive and you long press the map to create a route and move to active navigation the component that is listening to the long press and fetching the route needs to be in the `FreeDriveScreenBinder`.  When the route fetching is separated from the route line then there wouldn't be a need to include the route line component in `FreeDriveScreenBinder`.

**Comment by abhishek1508** [comment]
> > This looks good as it abstracts out fetching route from `DropInRouteLine` component. Only thing I am trying to wrap my head around is why do we need this component in `FreeDriveScreenBinder`? It should only be in `RoutePreviewScreenBinder` and `ActiveGuidanceScreenBinder`.
> 
> It was always there. The work here is to expose the route line options externally so they can be controlled by external developers. The work to refactor the route fetching will be in a different PR. If you're in free drive and you long press the map to create a route and move to active navigation the component that is listening to the long press and fetching the route needs to be in the `FreeDriveScreenBinder`. When the route fetching is separated from the route line then there wouldn't be a need to include the route line component in `FreeDriveScreenBinder`.

Yeah I just saw that route fetching logic is still in `DropInRouteLine`. Makes sense to have it in FD for now

**Review by abhishek1508 — APPROVED** [review]

**Pull request merged**